regnety/utils/tfrecords_utils.py

```diff
@@ -0,0 +1,184 @@
+
+    image_str = tf.io.encode_jpeg(image_tensor)
+
+    assert len(image_tensor.shape) == 3
+
+    return image_str
+
+
+def make_example(image_str, filepath, label, synset):
+
+    try:
+        example = tf.train.Example(
+            features=tf.train.Features(
+                feature={
+                    "image": _bytes_feature(image_str),
+                    "filename": _bytes_feature(
+                        os.path.basename(filepath)
+                    ),
+                    "label": _int64_feature(label),
+                    "synset": _bytes_feature(synset),
+                }
+            )
+        )
+    except:
+        example = tf.train.Example(
+            features=tf.train.Features(
+                feature={
+                    "image": _bytes_feature(image_str),
+                    "filename": _bytes_feature(
+                        os.path.basename(filepath)
+                    ),
+                    "label": _int64_feature(label),
+                    "synset": _bytes_feature(synset),
+                }
+            )
+        )
+    return example
+
+
+def make_single_tfrecord(
+    chunk_files, chunk_synsets, chunk_labels, output_filepath
+):
+
+    with tf.io.TFRecordWriter(output_filepath) as writer:
+        for i in range(len(chunk_files)):
+            image_str, height, width = make_image(chunk_files[i])
+            label = chunk_labels[i]
+            synset = chunk_synsets[i]
+
+            example = make_example(
+                image_str, height, width, chunk_files[i], label, synset
+            )
+
+            writer.write(example.SerializeToString())
+    writer.close()
+
+
+def make_tfrecs(
+    dataset_base_dir=None,  # example: home/imagenet/train
+    output_dir=None,  # example: home/imagenet_tfrecs
+    file_prefix=None,  # example: file_prefix = 'train' makes all files look like: train_0000_of_num_shards
+    synset_filepath=None,
+    batch_size=1024,
+):
+
+    """Driver function"""
+
+
+
+    images, labels, synsets = get_files(dataset_base_dir, synset_filepath)
+
+    filepaths_ds = tf.data.Dataset.from_tensor_slices([
+        (images[i]) for i in range(len(labels))
+    ])
+
+
+    images_ds = filepaths_ds.map(lambda filepath: make_image(filepath))
+    labels_ds = tf.data.Dataset.from_tensor_slices(labels)
+    synsets_ds = tf.data.Dataset.from_tensor_slices(synsets)
+    filepaths_ds = filepaths_ds.map(lambda filepath: tf.strings.split(filepath, '/')[-1])
+    
+    ds = tf.data.Dataset.zip((images_ds, filepaths_ds, labels_ds, synsets_ds))  
+    ds = ds.batch(batch_size, drop_remainder = False)
+    
+    num_shards = int(math.ceil(len(images) / batch_size))
+
+    shard = 0
+    for (image_str, filename, label, synset) in ds:
+        
+        output_filepath = os.path.join(
+            output_dir, file_prefix + "_%.4d_of_%.4d.tfrecord" % (shard, num_shards)
+        )
+        print("Writing %d of %d shards"%(shard, num_shards))
```

Comment:
Please be consistent with spacing (eg. either put spaces around the `%` sign as above, or remove them above).

Reply:
Yes